services/provider/server/server.go

```diff
@@ -799,6 +803,19 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 				rbdStorageClassData["encryptionKMSID"] = storageRequest.Spec.EncryptionMethod
 			}
 
+			blockPool := &rookCephv1.CephBlockPool{}
+			err = s.client.Get(ctx, types.NamespacedName{Name: rns.Spec.BlockPoolName, Namespace: s.namespace}, blockPool)
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "failed to get %s CephBlockPool. %v", blockPool.Name, err)
+			}
+
+			// SID for RamenDR
+			storageID := util.CalculateMD5Hash(fmt.Sprintf("%v-%v-%v",
+				cephCluster.Status.CephStatus.FSID,
+				blockPool.Status.PoolID,
+				rns.Name),
+			)
```

Comment:
Don't use string concatenation for producing ids. 
Use a struct (embedded struct) or array to represent the tuple - preferably a struct because you don't need to worry about ordering. Then serialize it to a well-know serialisation (either JSON or YAML). And lastly calculate the hash for the serialized string

Reply:
used a struct for producing the ids. The CalculateMD5Hash handle the serialisation to json and calculates the hash for the serialized string